### PR TITLE
Copr: Implements `take_scanned_range` for `BatchExecutorRunner`

### PR DESCRIPTION
**Pull request by mapleFU**
###  What have you changed?

This is a sub pr for #5361 . This pr:
- [x] Implements take_scanned_range for BatchExecutorRunner 

###  What is the type of the changes?

Pick one of the following and delete the others:

Implements take_scanned_range for BatchExecutorRunner .

###  How is the PR tested?

Updated integration test in #5361 .

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

###  Refer to a related PR or issue link (optional)

#5361

### CODE REVIEW

File: benches/coprocessor_executors/util/fixture.rs [hunk 1]
```diff
@@ -361,6 +361,12 @@ impl BatchExecutor for BatchFixtureExecutor {
     fn collect_storage_stats(&mut self, _dest: &mut Self::StorageStats) {
         // Do nothing
     }
+
+    #[inline]
+    fn take_scanned_range(&mut self) -> IntervalRange {
+        // TODO: pass values
```
Comment (breeswish) [inline benches/coprocessor_executors/util/fixture.rs]: unreachable should be enough

File: components/tidb_query/src/batch/executors/util/mock_executor.rs [hunk 2]
```diff
@@ -41,4 +42,10 @@ impl BatchExecutor for MockExecutor {
     fn collect_storage_stats(&mut self, _dest: &mut Self::StorageStats) {
         // Do nothing
     }
+
+    // TODO: make it fit next_batch
+    fn take_scanned_range(&mut self) -> IntervalRange {
+        // Do nothing
+        ("0", "1").into()
```
Comment (breeswish) [inline components/tidb_query/src/batch/executors/util/mock_executor.rs]: unreachable

File: components/tidb_query/src/batch/executors/util/scan_executor.rs [hunk 3]
```diff
@@ -122,6 +122,10 @@ impl<S: Storage, I: ScanExecutorImpl> ScanExecutor<S, I> {
         // Not drained
         Ok(false)
     }
+
+    pub(crate) fn take_scanned_range(&mut self) -> IntervalRange {
```
Comment (breeswish) [inline components/tidb_query/src/batch/executors/util/scan_executor.rs]: why?

**Comment by lonng** [comment]
> The rest LGTM. BTW why is this PR WIP?

See the description, he want to construct some unit tests.

**Comment by breeswish** [comment]
Any update? Is this PR still WIP?

**Pull request merged**